orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/DeadMessageHandler.kt

```diff
@@ -17,16 +17,19 @@
 package com.netflix.spinnaker.orca.q.handler
 
 import com.netflix.spinnaker.orca.ExecutionStatus.TERMINAL
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository
 import com.netflix.spinnaker.orca.q.*
 import org.slf4j.LoggerFactory
-import org.springframework.stereotype.Component
 
-@Component open class DeadMessageHandler {
+open class DeadMessageHandler(
+  private val executionRepository: ExecutionRepository
+) {
   private val log = LoggerFactory.getLogger(javaClass)
 
   open fun handle(queue: Queue, message: Message) {
     log.error("Dead message: $message")
     when (message) {
+      is CompleteExecution -> executionRepository.updateStatus(message.executionId, TERMINAL)
```

Comment:
It's possible we need some other conditions here but I think it would be nice to get this fix in to reduce churn in the metrics we're seeing currently.